docs/proposals/20250722-openstackclusteridentity.md

```diff
@@ -0,0 +1,213 @@
+
+#### Story 3: Multi-Region Setup
+As an administrator managing clusters across multiple OpenStack regions, I want to create region-specific cluster identities with appropriate credentials and allow tenants to use different regional credentials based on their needs.
+
+### API Design
+
+#### New OpenStackClusterIdentity Resource (Cluster-scoped)
+
+```go
+type OpenStackClusterIdentity struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    Spec   OpenStackClusterIdentitySpec   `json:"spec,omitempty"`
+    Status OpenStackClusterIdentityStatus `json:"status,omitempty"`
+}
+
+type OpenStackClusterIdentitySpec struct {
+    // SecretRef references the secret containing OpenStack credentials
+    SecretRef OpenStackCredentialSecretReference `json:"secretRef"`
+
+    // AllowedNamespaces defines which namespaces can use this identity
+    // +optional
+    AllowedNamespaces []string `json:"allowedNamespaces,omitempty"`
+
+    // NamespaceSelector selects allowed namespaces via labels
+    // +optional
+    NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+}
+
+type OpenStackCredentialSecretReference struct {
+    Name      string `json:"name"`
+    Namespace string `json:"namespace"`
+}
+
+type OpenStackClusterIdentityStatus struct {
+    Ready bool `json:"ready"`
+    Conditions clusterv1.Conditions `json:"conditions,omitempty"`
+}
+
+type OpenStackClusterIdentityReference struct {
+    Name string `json:"name"`
+}
+```
+
+**Enhanced OpenStackCluster:**
+```go
+type OpenStackClusterSpec struct {
+    // ... existing fields unchanged ...
+
+    // ClusterIdentityRef references a cluster-scoped identity (NEW)
+    // Takes precedence over IdentityRef if specified
+    // +optional
+    ClusterIdentityRef *OpenStackClusterIdentityReference `json:"clusterIdentityRef,omitempty"`
+
+    // IdentityRef references namespace-local secret (EXISTING - unchanged)
+    // +kubebuilder:validation:Required  
+    IdentityRef OpenStackIdentityReference `json:"identityRef"`
+}
+```
+
+### Implementation Details
+
+#### Credential Resolution Logic
+The scope factory will implement dual-path credential resolution:
+
+```go
+func (f *providerScopeFactory) resolveCredentials(obj infrav1.IdentityRefProvider) {
+    // Priority 1: Check for cluster identity reference
+    if clusterRef := obj.GetClusterIdentityRef(); clusterRef != nil {
+        return f.newScopeFromClusterIdentity(clusterRef)
+    }
+    
+    // Priority 2: Fall back to existing namespace identity behavior
+    return f.newScopeFromNamespaceIdentity(obj.GetIdentityRef())
+}
+```
+
+**Key Edge Cases:**
+- Both references specified: Use `clusterIdentityRef`, log warning
+- Identity deletion: Clusters show degraded status, don't fail
+- Permission changes: Dynamic re-validation during reconciliation
```

Comment:
What permissions are you referring to here?

Reply:
Clarified in docs.